src/Simplex_tree/include/gudhi/Simplex_tree/filtration_value_utils.h

```diff
@@ -0,0 +1,117 @@
+ *    Author(s):       Hannah Schreiber
+ *
+ *    Copyright (C) 2025 Inria
+ *
+ *    Modification(s):
+ *      - YYYY/MM Author: Description of the modification
+ */
+
+#ifndef SIMPLEX_TREE_FILTRATION_VALUE_UTILS_H_
+#define SIMPLEX_TREE_FILTRATION_VALUE_UTILS_H_
+
+#include <limits> //std::numeric_limits
+#include <cmath>  //std::isnan
+
+#include <gudhi/Simplex_tree/serialization_utils.h>
+
+namespace Gudhi {
+
+namespace simplex_tree {
+
+/**
+ * @ingroup simplex_tree
+ * @brief Returns `Filtration_value(0)` when converted to `Filtration_value`.
+ */
+inline struct empty_filtration_value_t {
+  template <class T> operator T() const { return T(0); }
+} empty_filtration_value;
+
+}  // namespace simplex_tree
+
+/**
+ * @ingroup simplex_tree
+ * @brief Given two filtration values at which a simplex exists, stores in the first value the minimal union of births
+ * generating a lifetime including those two values.
+ * This is the overload for when @ref FiltrationValue is an arithmetic type, like double, int etc.
+ * Because the filtration values are totally ordered then, the union is simply the minimum of the two values.
+ *
+ * NaN values are not supported.
+ */
+template <typename Arithmetic_filtration_value>
+bool unify_lifetimes(Arithmetic_filtration_value& f1, Arithmetic_filtration_value f2)
+{
+  if (f2 < f1){
+    f1 = f2;
+    return true;
+  }
+  return false;
+}
+
+/**
+ * @ingroup simplex_tree
+ * @brief Given two filtration values, stores in the first value the lowest common upper bound of the two values.
+ * If a filtration value has value `NaN`, it should be considered as the lowest value possible.
+ * This is the overload for when @ref FiltrationValue is an arithmetic type, like double, float, int etc.
+ * Because the filtration values are totally ordered then, the upper bound is always the maximum of the two values.
+ */
+template <typename Arithmetic_filtration_value>
+bool intersect_lifetimes(Arithmetic_filtration_value& f1, Arithmetic_filtration_value f2)
+{
+  if constexpr (std::numeric_limits<Arithmetic_filtration_value>::has_quiet_NaN) {
+    if (std::isnan(f1)) {
+      f1 = f2;
+      return !std::isnan(f2);
+    }
+
+    // Computes the max while handling NaN as lowest value.
+    if (!(f1 < f2)) return false;
+
+    f1 = f2;
+    return true;
+  } else {
+    // NaN not possible.
+    if (f1 < f2){
+      f1 = f2;
+      return true;
+    }
+    return false;
+  }
+}
+
+/**
+ * @ingroup simplex_tree
+ * @brief Serialize the given value and insert it at start position using
+ * @ref Gudhi::simplex_tree::serialize_trivial "".
+ * 
+ * @param[in] value The value to serialize.
+ * @param[in] start Start position where the value is serialized.
+ * @return The new position in the array of char for the next serialization.
+ * 
+ * @warning It is the user's responsibility to provide a pointer to a buffer with enough memory space.
+ */
+template <typename Trivial_filtration_value>
```

Comment:
There is the question of whether we should constrain this overload (enable_if, maybe with is_arithmetic, maybe with something else). I guess that can wait.

Reply:
Now that they are marked private, I guess that it is less important now. In the mean time, I added a description of the template parameter at least.